File: week4_story/TheWordMirror.cpp
/**
 * @filename   : TheWordMirror.cpp
 * @description: Check whether a word is a palindrome using recursion.
 * @author     : Ehtesham Nawaz
 */

#include <iostream>
using namespace std;

bool checkPalindrome(int left, int right, const string &word)
{
    if (left >= right)
        return true;

    if (word[left] != word[right])
        return false;

    return checkPalindrome(left + 1, right - 1, word);
}

int main()
{
    string word = "racecar";

    cout << (checkPalindrome(0, word.size() - 1, word) ? "True" : "False");
    return 0;
}

File: Story 4/TheWordMirror.java
/**
 * <p>This class solves The Word Mirror problem to check whether the string is pallindrome or not using recursion) </p>
 *
 * @Author- Aaditya Jain
 */

import java.util.Scanner;

public class TheWordMirror {
    static boolean checkPalindrome(String inputString) {
        if (inputString.length() <= 1)
            return true;
        if (inputString.charAt(0) != inputString.charAt(inputString.length() - 1))
            return false;
        return checkPalindrome(inputString.substring(1, inputString.length() - 1));
    }

    public static void main(String[] args) {
        Scanner scannerobj = new Scanner(System.in);
        System.out.println("Enter the string to check whether it is palindrome or not  ");
        String inputString = scannerobj.next();
        System.out.println(checkPalindrome(inputString));

    }
}
